### DISCUSSION TITLE
How to get wrapper for element outside mounted component?

### DISCUSSION THREAD

**Discussion by joerattz** [vuejs/test-utils#1738] [General]
Some components (like menus, dropdowns, dialogs, etc.) attach off the body element and outside the mounted component.  I can't find any way with VueTestUtils to obtain a wrapper to those so I can trigger a click event, like on a menu item.

What am I missing?

**Comment by freakzlike** [comment]
I got two possible solutions for this:
1. Create a wrapper component which holds the required outer elements. The template could look like `<body><your-menu /></body>`
2. Use the [`attachTo`](https://test-utils.vuejs.org/api/#attachto) mount option

I don't know if this is working four your use case

**Reply by joerattz** [reply]
Thanks for responding.  I don't understand how that would work for me because I'm testing my custom component that contains a menu and that menu teleports to the body element, outside my custom component.  If I mount my custom component, even using attach, no VueTestUtils API is going to allow me to get a reference to an element outside my mounted component, right? 

Even if I mount my component with attachTo and put it in the body, and the menu gets teleported to the body, I would end up with this:

```
<body>
  <my-custom-component />
  <menu />
</body>
```

So I still can't think of a way to get a wrapper for that menu with any VTU API?

**Reply by freakzlike** [reply]
Try Solution 1. When the template of your component looks like this:
```html
<teleport to="body">
  <menu />
</teleport>
```

You should be able to get the wrapper for the menu with:
```js
const wrapper = mount({
  template: '<body><my-custom-component /></body>'
})

const menu = wrapper.findComponent(Menu)
```

**Reply by joerattz** [reply]
My component doesn't contain the teleport element.  A third-party component (a menu component) contains it.

I suspect part of my problem is perhaps I'm trying to test something that is not part of my 'unit', i.e. my component.  I'm trying to test a component that is consumed by my component.  Perhaps that is or should be out of scope for a unit test.  However, in this case, that third-party component, a menu, will callback a function of mine that is implemented in my component and that my component reacts to so I'd like to test that when the menu is selected my component does what it needs to do correctly.

**Reply by rivaldi8** [reply]
It seems the way to do this is by stubbing the `Teleport` component, like in the example at the end of [Testing Teleport page](https://test-utils.vuejs.org/guide/advanced/teleport.html):

```js
test('teleport', async () => {
  const wrapper = mount(Navbar, {
    global: {
      stubs: {
        teleport: true
      }
    }
  })
})
```
